regulations/static/regulations/js/source/views/comment/comment-view.js

```diff
@@ -11,32 +11,24 @@ require('prosemirror/dist/markdown');
 
 var CommentModel = require('../../models/comment-model');
 var CommentEvents = require('../../events/comment-events');
+var AttachmentView = require('../../views/comment/attachment-view');
 var comments = require('../../collections/comment-collection');
 
 function getUploadUrl(file) {
   var prefix = window.APP_PREFIX || '/';
   return $.getJSON(
     prefix + 'comments/attachment',
-    {size: file.size}
+    {size: file.size, type: file.type || 'application/octet-stream'}
   ).then(function(resp) {
     return resp;
   });
 }
 
-function readFile(file) {
-  var deferred = $.Deferred();
-  var reader = new FileReader();
-  reader.onload = function() {
-    deferred.resolve(reader.result);
-  };
-  reader.readAsBinaryString(file);
-  return deferred;
-}
-
 var CommentView = Backbone.View.extend({
   events: {
-    'change input[type="file"]': 'addAttachment',
-    'click .queue-item': 'clearAttachment',
+    'change input[type="file"]': 'addAttachments',
+    'dragenter input[type="file"]': 'highlightDropzone',
+    'dragleave input[type="file"]': 'unhighlightDropzone',
```

Comment:
Depends on the desired behavior. If we want to highlight the dropzone on hovering the pointer over it, then yes. This implementation is a little different, highlighting the dropzone when something's dragged over it. That was the more intuitive behavior to me, but if highlighting on hover is right, we can simplify the code a bit.